### DISCUSSION TITLE
Facing challenges in sysbuild flashing using west flash on nucleo_u575zi_q

### DISCUSSION THREAD

**Discussion by PragatiGarg-eaton** [zephyrproject-rtos/zephyr#59208] [Q&A] [answered]
Hi, 

I am working on building and flashing MCUBOOT+ application using sysbuild on nucleo_u575zi_q.

- nucleo_u575zi_q board is configured to be flashed using west STM32CubeProgrammer runner. 
- I am facing challenges in west flashing. After building the signed application along with MCUBOOT using sysbuild command, it is performing mass erase every time it flashes either mcuboot or signed image.
- And ideally, when invoking west flash on a build consisting of multiple images, each image is flashed in sequence without any chip erase.

west flash mcuboot:
![westflash_1](https://github.com/zephyrproject-rtos/zephyr/assets/107679628/a6e4125a-4dd3-4227-9059-ee95de965ec2)

west flash of signed image:
![westflash_2](https://github.com/zephyrproject-rtos/zephyr/assets/107679628/03d416ab-c854-4d1f-a202-1ec9fb7d8dfe)
![westflash_3](https://github.com/zephyrproject-rtos/zephyr/assets/107679628/a06b656d-953b-43b0-834e-944ea18186c3)

Commands used to build and flash:
build command: `west build -b nucleo_u575zi_q -p --sysbuild -- -DCONFIG_<mcuboot>=<y> --DCONFIG_<local signing>=<y>`
flash: `west flash`
Question: How can we disable the mass erase option with every west flash here?

OS: Windows
zephyr version: 3.2.0
STM32CubeProgrammer: 2.11.0

Links:
https://docs.zephyrproject.org/latest/build/sysbuild/index.html
https://docs.zephyrproject.org/3.0.0/boards/arm/nucleo_u575zi_q/doc/nucleou575zi_q.html#programming

**Comment by erwango** [comment, marked as answer]
Erase happens because of this line in board's board.cmak file: board_runner_args(stm32cubeprogrammer "--erase" "--port=swd" "--reset-mode=hw")
Removing the --erase should solve your issue

**Comment by PragatiGarg-eaton** [comment]
Superb! it's working now. Thanks.
